### PR TITLE
dev: Use taplo to format TOML files of the project

### PR DESCRIPTION
**Pull request by ApereLieZ**
# Pull Request type

Please add the labels corresponding to the type of changes your PR introduces:

- Code style update (formatting, renaming)
- Refactoring (no functional changes, no API changes)

## What is the current behavior?

Resolves: #607

## What is the new behavior?

- Add instructions on how to run taplo locally
- Add a formatting rules file for universal formatting
- Add a workflow in the CI to check the formal of TOML files and fail if not valid
- add husky hooks with TOML file validation (need to run hooks once)

## Does this introduce a breaking change?

Yes
You must check the TOML files before pushing, otherwise ci/cd will fail.
To prevent this behavior, enable husky hooks
 
```bash
cargo test 
```

### CODE REVIEW

**Review by EvolveArt — CHANGES_REQUESTED** [review]
As said in madara channel :
> couldn't we just use the taplo rust crate. Not a huge fan of having to add npm dependencies at the root of the project..

We should rather use [cargo-husky](https://github.com/rhysd/cargo-husky) there imo

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -78,3 +78,10 @@ jobs:
         run: |-
           cd tests
           npm run test
+  toml-lint:
```
Comment (EvolveArt) [inline .github/workflows/test.yml:81]: add spacing here

**Comment by EvolveArt** [comment]
@ApereLieZ why does it still use `npx @taplo/cli` in the CI ? I don't get why we cannot just use the [taplo](https://crates.io/crates/taplo/0.12.0) crate ?

**Comment by ApereLieZ** [comment]
> @ApereLieZ why does it still use `npx @taplo/cli` in the CI ? I don't get why we cannot just use the [taplo](https://crates.io/crates/taplo/0.12.0) crate ?

From my measurements in terms of runtime, using npx is noticeably faster than using crate

**Comment by EvolveArt** [comment]
> > @ApereLieZ why does it still use `npx @taplo/cli` in the CI ? I don't get why we cannot just use the [taplo](https://crates.io/crates/taplo/0.12.0) crate ?
> 
> @EvolveArt From my measurements in terms of runtime, using npx is noticeably faster than using crate

But then no need to `cargo add taplo-cli` in this case ?

**Review by EvolveArt — APPROVED** [review]
lgtm

**Comment by ApereLieZ** [comment]
@EvolveArt 
Done

**Pull request merged**